### ISSUE TITLE
Issue with training non-english (Kannada) pre-trained model for classification

### ISSUE THREAD

**Issue by SushmaBhat** [facebookresearch/fastText#709]
I am trying to create a text classifier for Kannada Language (South Indian Language) using fasttext pre-trained model. wiki.kn.vec. The train data set is a product review dataset in kannada. I am using the below command to train the model using the above Kannada pre-trained word2vec data. 

./fasttext supervised -input train_kan.txt -output model_kan -epoch 250 -lr 1.0 -verbose 2 -pretrainedVectors wiki.kn.vec

However the generated supervised model is empty and the output of the above command is:
Read 0M words
Number of words:  364
Number of labels: 2

What could be possibly going wrong here? How do I mention that the dataset provided is UTF-8? Any pointers? I checked the previous forum questions related to non-English and could not find an answer. Let me know if anyone has faced this issue.

**Comment by infinite-Joy** [comment]
there might be an issue with the files. can you please share a head of the training files

    head train_kan.txt

**Comment by SushmaBhat** [comment]
I have been successful to create a classifier without the pre-trained vector. here is a sample.

__label__1	u"ಈ ಉತ್ಪನ್ನವನ್ನು ಖರೀದಿಸಲು ನೀವು ಚಿಂತಿಸುವುದಿಲ್ಲ"
__label__1	u"ಉತ್ಪನ್ನವು ಡಿವಿಡಿಯಿಂದ ಉನ್ನತ ಗುಣಮಟ್ಟದ ಅಪ್ ಸ್ಕೇಲಿಂಗ್ ವೀಡಿಯೊ ಪ್ರೊಸೆಸರ್ಗಳ ಅಸಾಧಾರಣ ಚಿತ್ರದ ಗುಣಮಟ್ಟವನ್ನು ಉತ್ಪಾದಿಸುತ್ತದೆ"
__label__0	u"ಇದು ಹೊಸ ವೈಶಿಷ್ಟ್ಯವನ್ನು ಒದಗಿಸುವುದಿಲ್ಲ"
__label__1	u"ಈ ಉತ್ಪನ್ನವನ್ನು ಖರೀದಿಸಿದ ನಂತರ ನೀವು ವಿಷಾದ ಮಾಡುವುದಿಲ್ಲ"
__label__1	u"ಚೆನ್ನಾಗಿ ಕಾರ್ಯನಿರ್ವಹಿಸುತ್ತದೆ"
__label__1	u"ನಾನು ಈ ಉತ್ಪನ್ನವನ್ನು ಖರೀದಿಸುತ್ತೇನೆ"
__label__0	u"ನಾನು ಈ ಉತ್ಪನ್ನವನ್ನು ಶಿಫಾರಸು ಮಾಡುವುದಿಲ್ಲ"

**Comment by SushmaBhat** [comment]
The issue was with the dimension -dim 300. While specifying the pre-trained model, I had to give the dimension of the original data that was used to create it

**Comment by EdouardGrave** [comment]
Hi @SushmaBhat,

Thank you for raising this issue!

It seems that you were able to fix the problem by changing the dimension of the model. We will make the error message more clear for potential future users.

Best,
Edouard.

**Comment by SushmaBhat** [comment]
Thank you
